conbench/cli.py

```diff
@@ -81,6 +86,61 @@ def _to_cli_name(name):
     return f"--{name.replace('_', '-')}"
 
 
+def log_timing_report_across_results(results):
+    # Find tags that are common among all results. That makes sense only when
+    # there is more than one result.
+    if len(results) == 1:
+        common_tag_keys = []
+    else:
+        common_tags = set.intersection(*tuple(set(r["tags"].items()) for r in results))
+        common_tag_keys = [t[0] for t in common_tags]
+        log.info("common tag keys across results: %s", common_tag_keys)
+
+    # For each result, assemble a human-readable string which represents the
+    # case/scenario from the tags. Also assemble a human-readable string that
+    # reports the timing info for this scenario.
+    tagstrings = []
+    timestrings = []
+
+    for r in results:
+        tags = r["tags"]
+
+        # That represents the case/senario. Note that `tags` has been enriched
+        # with key/value pairs beyond the case -- filter these out again.
+        tagstring = ", ".join(
+            f"{k}: {tags[k]}" for k in sorted(tags.keys()) if k not in common_tag_keys
+        )
+
+        # Assume that stdev=0 for now means there is only one sample.
+        if r["stats"]["stdev"] == 0:
+            timestring = f"{sigfig.round(r['stats']['data'][0], 2)} s"
+        else:
+            # Assume that there are at least three samples. Standard deviation
+            # is known, calc standard error of the mean:
+            mean = r["stats"]["mean"]
+            stdev = r["stats"]["stdev"]
+            min = r["stats"]["min"]
```

Comment:
There was a fairly recent change so that summary stats like these don't need to be supplied by the user, and the server will do so from `data` before inserting the database record. (This is nice because it ensures consistency in calculation regardless of what is actually running benchmarks.) I think this runner is still calculating them client-side so this will work, but if we port this elsewhere, we can't depend on these existing.